### PR TITLE
feat: integrate data catalog

### PR DESCRIPTION
**Pull request by RanbirAulakh**
**Issue #, if available:** n/a

### Notes
- Integrate data catalog

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/osml/stac_catalog/sc_dataplane.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    // Environment variables for the Lambda function
+    const environment = {
+      STAC_FASTAPI_TITLE: "stac-fastapi-opensearch",
+      STAC_FASTAPI_DESCRIPTION: "A STAC FastAPI with an OpenSearch backend",
+      STAC_FASTAPI_VERSION: "3.0.0a1",
+      RELOAD: "true",
+      ENVIRONMENT: "local",
+      WEB_CONCURRENCY: "10",
+      ES_HOST: "opensearch",
+      ES_PORT: "9202",
```
Comment (drduhe) [inline lib/osml/stac_catalog/sc_dataplane.ts:114]: Make this a config param

File: lib/osml/stac_catalog/sc_dataplane.ts [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+      enforceHttps: true,
+      encryptionAtRest: {
+        enabled: true
+      },
+      vpc: props.osmlVpc.vpc,
+      capacity: {
+        dataNodes: this.config.OS_DATA_NODES
+      },
+      vpcSubnets: [props.osmlVpc.selectedSubnets],
+      removalPolicy: RemovalPolicy.DESTROY,
```
Comment (drduhe) [inline lib/osml/stac_catalog/sc_dataplane.ts:153]: this should be flagged for the isProd setting.

File: lib/osml/stac_catalog/sc_dataplane.ts [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+      zoneAwareness: {
+        enabled: true,
+        availabilityZoneCount: this.config.A2Z_ZONES
```
Comment (drduhe) [inline lib/osml/stac_catalog/sc_dataplane.ts:156]: A2Z?

**Review by drduhe — CHANGES_REQUESTED** [review]

File: lib/osml/stac_catalog/sc_dataplane.ts (hunk 2 above)
Comment (RanbirAulakh) [reply lib/osml/stac_catalog/sc_dataplane.ts]: fixed

File: lib/osml/stac_catalog/sc_dataplane.ts (hunk 3 above)
Comment (RanbirAulakh) [reply lib/osml/stac_catalog/sc_dataplane.ts]: fixed

**Review by drduhe — APPROVED** [review]
LGTM

**Pull request merged**